### ISSUE TITLE
Training loss curve like stairs.

### ISSUE THREAD

**Issue by yuanzhedong** [tatsu-lab/stanford_alpaca#236]
I'm trying to rerun the training and the loss curve dramatic drops after each epoch, does anyone know why? Below is a loss curve I trained with 3 epochs

<img width="333" alt="image" src="https://user-images.githubusercontent.com/5069709/233734443-394a8f44-da70-41af-9bdb-359b90a97d8c.png">

**Comment by yuanzhedong** [comment]
The cmd I'm using
```
torchrun --nproc_per_node=8 train.py \
    --model_name_or_path ./llama-7b-hf \
    --data_path ./alpaca_data.json \
    --bf16 True \
    --output_dir /workspace/code/output \
    --num_train_epochs 3 \
    --per_device_train_batch_size 4 \
    --per_device_eval_batch_size 4 \
    --gradient_accumulation_steps 8 \
    --evaluation_strategy "no" \
    --save_strategy "steps" \
    --save_steps 2000 \
    --save_total_limit 1 \
    --learning_rate 2e-5 \
    --weight_decay 0. \
    --warmup_ratio 0.03 \
    --lr_scheduler_type "cosine" \
    --logging_steps 1 \
    --fsdp "full_shard auto_wrap" \
    --fsdp_transformer_layer_cls_to_wrap "LlamaDecoderLayer" \
    --tf32 True
    --report_to wandb
```

**Comment by WooKimm** [comment]
Hi! Have you solved the problem?

**Comment by echoht** [comment]
hi can you share the scale of training loss?
The following is mine. The loss of the beginning is very big!!  I use deepspeed and gradient checkpointing.
![image](https://github.com/tatsu-lab/stanford_alpaca/assets/48375360/4fe1ecef-3030-483a-8d33-cb2df888da0b)

**Comment by yuanzhedong** [comment]
I think [this](https://github.com/huggingface/transformers/issues/18730#issuecomment-1229769519) may explain why. Basically the dataset is too small and it kind of remember the dataset, so for the second time the model sees the same data again the loss will dive.

**Comment by echoht** [comment]
I have the question about my beginning loss, which is about 8. This phenomenon is very strange. 
What are the factors that lead to this kind of big loss?

**Comment by yuanzhedong** [comment]
<img width="1433" alt="image" src="https://github.com/tatsu-lab/stanford_alpaca/assets/5069709/51570d78-dbaf-4ce4-8c72-52da0d99241b">

This is the loss curve I have.

**Comment by ftgreat** [comment]
Same loss curve.

**Comment by zycheiheihei** [comment]
> <img alt="image" width="1433" src="https://user-images.githubusercontent.com/5069709/237271945-51570d78-dbaf-4ce4-8c72-52da0d99241b.png">
> This is the loss curve I have.

@yuanzhedong  Hi, is this kind of loss curve normal or expected ? I am reproducing the finetuning of llama-7b and want to see whether there is something going wrong.
